docs/src/Groups/groups.md

```diff
@@ -22,6 +22,160 @@ Julia supports the following types of groups:
+```
+
+  **Example:**
+```jldoctest
+julia> perm(symmetric_group(6),[2,4,6,1,3,5])
+(1,2,4)(3,6,5)
+
+julia> cperm([1,2,3],4:7)
+(1,2,3)(4,5,6,7)
+
+julia> cperm([1,2],[2,3])
+(1,3,2)
+```
+
+At the moment, the input vectors of the function `cperm` need not to be disjoint.
+
+!!! warning
+    If the function `perm` is evaluated in a vector of integers without specifying the group `G`, then the returned value is an element of the AbstractAlgebra.jl type `Perm{Int}`. For this reason, if one wants a permutation of type `GAPGroupElem{PermGroup}`, one has to use the function `gap_perm`.
+
+
+Every permutation has always a permutation group as a parent. Two permutations coincide if, and only if, they have the same parent and they move the same points.
+```jldoctest
+julia> G=symmetric_group(5);
+julia> x=cperm([1,2]);
+julia> y=cperm(G,[1,2]);
+julia> x==y
+false
+```
+In the example above, `x` and `y` are different because `x` belongs to `Sym(2)`, while `y` belongs to `Sym(5)`.
+
+
+If `G` is a group and `x` is a permutation, it is possible to set `G` as parent of `x` simply typing `G(x)`. This returns the permutation `x` as element of `G` (or ERROR if `x` does not embed into `G`).
+```@repl oscar
+G=symmetric_group(5);
+x=cperm([1,2,3]);
+y=G(x);
+parent(x)
+parent(y)
+```
+
+The function `listperm` works in the opposite way with respect to `perm`: given a permutation `x`, it returns the vector `[x(1), ... , x(n)]`, where `n` is the degree of `parent(x)`. If `x` and `y` are as in the example above, we get
+```@repl oscar
+(x,y) = (cperm([1,2,3]), cperm(symmetric_group(5),[1,2,3])) # hide
+listperm(x)
+listperm(y)
+```
+
+
+#### Permutations as functions
+A permutation can be viewed as a function on the set `{1,...,n}`, hence it can be evaluated on integers.
+```jldoctest
+julia> x = cperm([1,2,3,4,5]);
+julia> x(2)
+3
+```
+This works also if the argument is not in the range `1:n`; in such a case, the output coincides with the input.
+
+!!! note
+    The multiplication between permutations works from the left to the right. So, if `x` and `y` are permutations and `n` is an integer, then `(x*y)(n) = (y(x(n))`, NOT `x(y(n))`.
+
+
+## Polycyclic groups
+
+Julia has the following functions that allow to generate polycyclic groups:
+```@docs
+abelian_group(::Type{T}, v::Vector{Int}) where T <: GAPGroup
+cyclic_group
+dihedral_group
+quaternion_group
+```
+!!! warning
+    The type need to be specified in the input of the function `abelian_group`, otherwise a group of type `GrpAbFinGen` is returned.
+
+The generators of a polycyclic group are displayed as `f1`, `f2`, `f3`, etc., and every element of a polycyclic group is displayed as product of such generators.
+
+  **Example:**
+```jldoctest
+julia> G=abelian_group(PcGroup, [2,4]);
+julia> G[1], G[2]
+(f1, f2)
+julia> G[2]*G[1]
+f1*f2
+```
+Anyway, recall that variables named `f1`, `f2`, etc. are not initialized! To get the generators of the group `G`, one has to refer to them as `G[1]`, `G[2]`, etc.
```

Comment:
```suggestion
Note that this does not define Julia variables named `f1`, `f2`, etc.! To get the generators of the group `G`, use `gens(G)`; for convenience they can also be accessed as `G[1]`, `G[2]`, etc.
```
"Anyway" is colloquial. And "recall" doesn't seem to apply (from where should this be recalled?). Also note that the point about "there are no Julia variables with those names by default" also applies to free groups and in general all finitely presented groups (of which pc groups in a sense are a special case).

At the same time, access to the generators is something that really concerns *all* groups, so perhaps it would be better to move that into its own section, and then just insert a reference link here. (Please look into the Documenter.jl documentation to learn how to insert cross links to other section)

Finally, you could also mention the possibility of writing `f1, f2 = gens(G)`.